internal/notifier/factory.go

```diff
@@ -66,19 +68,22 @@ type notifierMap map[string]factoryFunc
 type factoryFunc func(opts notifierOptions) (Interface, error)
 
 type notifierOptions struct {
-	URL               string
-	ProxyURL          string
-	Username          string
-	Channel           string
-	Token             string
-	Headers           map[string]string
-	CertPool          *x509.CertPool
-	Password          string
-	CommitStatus      string
-	ProviderName      string
-	ProviderNamespace string
-	SecretData        map[string][]byte
-	TokenCache        *pkgcache.TokenCache
+	URL                string
+	ProxyURL           string
+	Username           string
+	Channel            string
+	Token              string
+	Headers            map[string]string
+	CertPool           *x509.CertPool
+	Password           string
+	CommitStatus       string
+	ProviderName       string
+	ProviderNamespace  string
+	SecretData         map[string][]byte
+	ServiceAccountName string
+	TokenCache         *pkgcache.TokenCache
+	TokenClient        client.Client
+	Context            context.Context
```

Comment:
Let's also move `Context` to the beginning to match the Go idiom of `ctx` being always the first argument in a function